relation/src/gadgets/ecc/mod.rs

```diff
@@ -133,20 +129,20 @@ impl PointVariable {
 }
 
 // ECC related gates
-impl<F> PlonkCircuit<F>
-where
-    F: PrimeField,
-{
+impl<F: PrimeField> PlonkCircuit<F> {
     /// Return the witness point for the circuit
-    pub fn point_witness(&self, point_var: &PointVariable) -> Result<Point<F>, PlonkError> {
+    pub fn point_witness(&self, point_var: &PointVariable) -> Result<Point<F>, CircuitError> {
         self.check_point_var_bound(point_var)?;
         let x = self.witness(point_var.0)?;
         let y = self.witness(point_var.1)?;
         Ok(Point(x, y))
     }
 
     /// Add a new EC point (as witness) to the circuit
-    pub fn create_point_variable(&mut self, point: Point<F>) -> Result<PointVariable, PlonkError> {
+    pub fn create_point_variable(
+        &mut self,
+        point: &Point<F>,
```

Comment:
I think move the ownership of `point` (consume the input) is a good idea generally. 
IMO, the workflow generally goes like you construct a point through whatever step, then finally create a variable and from then on, you should only deal with `PointVar`. 
I can't think of a reason why would you still want the original `Point`. 

^^ maybe opinionated, would you agree? @zhenfeizhang @chancharles92

Reply:
The scenario I can think of is when the point is a public variable, then we'd still like keep the point so that we can later pass it as a public input. Do we also have an API `create_public_point_variable()`?

Reply:
Yes there's a `create_public_point_variable()`. And in the original codebase, it still consumes the ownership.

Reply:
> when the point is a public variable, then we'd still like keep the point so that we can later pass it as a public input. Do we also have an API create_public_point_variable()?

when a point is supposed to be a public input, you will consume it using `create_public_point_variable()`, usually (i mean most of the time) you won't create both a witness point and a PI point using the _same_ point. 
in the rare case that you do, you can always clone. 

> According to https://github.com/EspressoSystems/jellyfish/pull/107, can we still pass in the reference here?

that PR is about passing in `rng: impl RngCore + CryptoRng`, not other struct, so won't affect any decision here.